src/gpu/intel/jit/gemm/jit_gemm_pd.cpp

```diff
@@ -387,16 +389,33 @@ dim_t jit_gemm_pd_t::stride_binary(int idx, int stride) const {
     }
 }
 
-dim_t jit_gemm_pd_t::stride_scale(int idx, int arg) const {
-    const auto md = arg == DNNL_ARG_A ? desc()->b_desc : desc()->a_desc;
+dim_t jit_gemm_pd_t::eff_a_scale_stride(int idx) const {
+    dim_t stride = 1;
+    if (a_scale_md_.dims[idx] == 1) return 0;
+    for (int i = idx + 1; i < a_scale_md_.ndims; ++i) {
+        stride *= a_scale_md_.dims[i];
+    }
+    return stride;
+}
```

Comment:
Similarly for B strides
```suggestion
dim_t jit_gemm_pd_t::eff_a_scale_stride(int idx) const {
    gpu_assert(memory_desc_wrapper(a_scale_md_).is_plain()) << "Expected plain a_scale_md_";
    return a_scale_md_.format_desc.blocking.strides[idx];
}
```